### PR TITLE
Atomic Save for Setting

### PR DESCRIPTION
**Pull request by taooceros**
Atomic Save for Setting By

1. Create a Temp File
2. Write the new setting to the Temp File
3. Call `File.Replace` to replace the Temp File to the Setting File and create a backup
4. Delete the Temp File (Well If we want to can keep it there and use it as the second backup)

Interesting thing is: I also learn this strategy in my OS class.

Tested
- [x] Save Successfully create a backup file
- [x] If the original setting file is invalid, it will load the backup file

Should closes https://github.com/Flow-Launcher/Flow.Launcher/issues/641

Whether `File.Replace` is atomic is unclear to me due to discussion in https://stackoverflow.com/a/35128468/13906571. Though, if I understand the post correctly, it is safe to use if we always keep a backup, with the additional document in https://learn.microsoft.com/en-us/windows/win32/fileio/deprecation-of-txf?redirectedfrom=MSDN.

### CODE REVIEW

**Comment by taooceros** [comment]
> # @check-spelling-bot Report
> ### 🔴 Please review
> 
> See the [📂 files](1763/files/) view or the [📜action log](https://github.com/Flow-Launcher/Flow.Launcher/actions/runs/3863547131/jobs/6585801595#step:2:1) for details.
> #### Unrecognized words (1)
> 
> bak
> To accept ✔️ these unrecognized words as correct and remove the previously acknowledged and now absent words, run the following commands
> 
> To have the bot do this for you, reply quoting the following line: @check-spelling-bot apply [updates](https://github.com/Flow-Launcher/Flow.Launcher/actions/runs/3863547131/attempts/1).
> If the flagged items are 🤯 false positives

**Comment by taooceros** [comment]
Well I still don't understand how to let the bot apply the update🤣🤣

**Comment by taooceros** [comment]
> # @check-spelling-bot Report
> ### 🔴 Please review
> 
> See the [📂 files](1763/files/) view or the [📜action log](https://github.com/Flow-Launcher/Flow.Launcher/actions/runs/3863547131/jobs/6585801595#step:2:1) for details.
> #### Unrecognized words (1)
> 
> bak
> To accept ✔️ these unrecognized words as correct and remove the previously acknowledged and now absent words, run the following commands
> 
> To have the bot do this for you, reply quoting the following line: @check-spelling-bot apply [updates](https://github.com/Flow-Launcher/Flow.Launcher/actions/runs/3863547131/attempts/1).
> If the flagged items are 🤯 false positives

@check-spelling-bot apply [updates](https://github.com/Flow-Launcher/Flow.Launcher/actions/runs/3863547131/attempts/1).

**Comment by jjw24** [comment]
> > Well I still don't understand how to let the bot apply the update🤣🤣
> 
> I disabled it in settings because the author says its experimental. You can edit `spelling.yml` to enable it.

Ohh no wonder I wasnt able to as well 🤣🤣

**Comment by jjw24** [comment]
> > Well I still don't understand how to let the bot apply the update🤣🤣
> 
> I disabled it in settings because the author says its experimental. You can edit `spelling.yml` to enable it.

DW about it then since it's experimental.

File: Flow.Launcher.Infrastructure/Storage/JsonStorage.cs [hunk 1]
```diff
@@ -11,62 +12,73 @@ namespace Flow.Launcher.Infrastructure.Storage
...
+            if (!File.Exists(BackupFilePath))
+                return default;
+
+            try
             {
-                BackupOriginFile();
+                var data = JsonSerializer.Deserialize<T>(File.ReadAllText(BackupFilePath));
+                if (data != null)
+                {
+                    Log.Info($"|JsonStorage.Load|Load backup file {BackupFilePath} successfully");
```
Comment (jjw24) [inline Flow.Launcher.Infrastructure/Storage/JsonStorage.cs:72]: Should this message be failed to load settings.json, back up file loaded successfully?

Comment (jjw24) [reply Flow.Launcher.Infrastructure/Storage/JsonStorage.cs:72]:
```suggestion
                    Log.Info($"|JsonStorage.Load|Failed to load settings.json, {BackupFilePath} restored successfully");
```

**Review by jjw24 — APPROVED** [review]

**Pull request merged**